utils/src/main/kotlin/OkHttpClientHelper.kt

```diff
@@ -45,7 +46,7 @@ object OkHttpClientHelper {
     private const val MAX_CACHE_SIZE_IN_BYTES = 1024L * 1024L * 1024L
 
     private val client = buildClient()
-    private val clients = mutableMapOf<BuilderConfiguration, OkHttpClient>()
+    private val clients = ConcurrentHashMap<BuilderConfiguration, OkHttpClient>()
```

Comment:
While it probably makes no difference for our use cases, the current implementation is still not fully clean, even with a concurrent map, because it uses `getOrPut()` to access and populate the map. The Kotlin documentation for this function states:

_Note that the operation is not guaranteed to be atomic if the map is being modified concurrently._

So I think it is preferable to use `computeIfAbsent()` instead.